Create README.md for CTLs (#32529)
dotnet/sdk: src/Layout/redist/trustedroots/README.md

# Trusted X.509 Certificate Roots

This directory contains [Microsoft-maintained root certificate trust lists (CTL)](https://learn.microsoft.com/security/trusted-root/program-requirements) for [Code Signing (CS)](https://en.wikipedia.org/wiki/Code_signing), and [Time Stamping (TS)](https://en.wikipedia.org/wiki/Code_signing#Time-stamping). The .NET SDK uses these CTLs for [NuGet signing and validation](https://learn.microsoft.com/en-us/nuget/reference/signed-packages-reference), on Linux and macOS. They are also used in various Microsoft products. 

- [Code signing CTL](codesignctl.pem)
- [Timestamping CTL](timestampctl.pem)

The CTLs are stored in [PEM format](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail). 

## Behavior

[NuGet uses these CTLs as a fallback](https://github.com/dotnet/sdk/issues/25686) when an OS-provided CTL is not available.

> - Linux: The fallback CTL is typically used, however, an OS-provided CTL will be used if available (see `ca-certificates` section).

Reviewer: ```suggestion
- Linux: The fallback CTL is typically used; however, an OS-provided CTL will be used if available (see `ca-certificates` section).
```

> - macOS: The fallback CTL is always used since an OS-provided CTL is never available, when signed package verification is enabled (which is not the default).

Reviewer: ```suggestion
- macOS: The fallback CTL is always used when signed package verification is enabled, which is not the default since an OS-provided CTL is never available.
```

- Windows: The fallback CTL is never used since an OS-provided CTL is always available (via an OS API).

## Linux

On Linux, NuGet will first probe for a code signing system bundle (multi-PEM file) using a [list of well-known paths](https://github.com/dotnet/designs/blob/main/accepted/2021/signed-package-verification/re-enable-signed-package-verification-technical.md#linux). The first successful match will be used. If no match is found or if there are problems processing the system bundle, NuGet will use the fallback bundle.

> The timestamping CTL in the fallback bundle is always used. There doesn't seem to be any precedent for a timestamping-specific certificate bundle under `/etc/pki/ca-trust/extracted/pem`.

Reviewer: ```suggestion
The timestamping CTL in the .NET SDK is always used. There doesn't seem to be any precedent for a timestamping-specific certificate bundle under `/etc/pki/ca-trust/extracted/pem`.
```
